examples/supply_chain/common/balance_calculator.py

```diff
@@ -1,50 +1,39 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
-import collections
-from collections import defaultdict, namedtuple
-from typing import Dict, List, Tuple
+
+from collections import defaultdict, deque, Counter
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple
 
 import numpy as np
 
 from maro.simulator import Env
-
 from maro.simulator.scenarios.supply_chain.facilities.facility import FacilityInfo
 from maro.simulator.scenarios.supply_chain.objects import SupplyChainEntity
 from maro.simulator.scenarios.supply_chain.units.distribution import DistributionUnitInfo
 from maro.simulator.scenarios.supply_chain.units.storage import StorageUnitInfo
 
+from .utils import get_attributes, get_list_attributes
+
 
-ProductInfo = namedtuple(
-    "ProductInfo",
-    (
-        "unit_id",
-        "sku_id",
-        "node_index",
-        "storage_index",
-        "distribution_index",
-        "downstream_product_unit_id_list",
-        "product_index",
-        "consumer_index",
-        "seller_index",
-        "manufacture_index",
-    ),
-)
-
-
-FacilityLevelInfo = namedtuple(
-    "FacilityLevelInfo",
-    (
-        "unit_id",
-        "product_unit_id_list",
-        "storage_index",
-        "distribution_index",
-    ),
-)
+@dataclass
+class GlobalProductInfo:
+    unit_id: int
+    sku_id: int
+    node_index: int
+    storage_index: Optional[int]
+    distribution_index: Optional[int]
+    consumer_index: Optional[int]
+    seller_index: Optional[int]
+    manufacture_index: Optional[int]
+    downstream_product_id_list: List[int]
```

Comment:
Shall this one be changed to `sku_id`?

There are still many `product_id` in the project.

Reply:
product_id means product unit id here.